compiler/rustc_mir_transform/src/coverage/tests.rs

```diff
@@ -653,9 +653,7 @@ fn test_make_bcb_counters() {
         // BCB counter construction that those BCBs have spans.
         let bcb_has_coverage_spans = |bcb: BasicCoverageBlock| (1..=2).contains(&bcb.as_usize());
         let mut coverage_counters = counters::CoverageCounters::new(&basic_coverage_blocks);
-        coverage_counters
-            .make_bcb_counters(&mut basic_coverage_blocks, bcb_has_coverage_spans)
-            .expect("should be Ok");
```

Comment:
Since this code is changing anyway, I've taken the opportunity to also get rid of some unnecessary `&mut` references to the BCB graph that were left over from #114354.